kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt

```diff
@@ -48,6 +48,9 @@ class ClassBuilderDsl<T : KtObject>(
         require(!constructors.containsKey(constructor.parameterCount)) {
             "A constructor with ${constructor.parameterCount} argument(s) already exists."
         }
+        require(constructor.parameterCount <= CONSTRUCTOR_MAX_ARGS) {
+            "Cannot register a constructor with ${constructor.parameterCount} arguments, max argument count is $CONSTRUCTOR_MAX_ARGS"
+        }
```

Comment:
I don't remember anymore: why do we have a limit of 5 args per constructor?

Reply:
Constructor call is pretty much the same as method call. IIRC we can have more than 5 params, but it becomes less performant because there is unpacking.

Reply:
Ok I will add that as an explanation to the docs